frame/staking-rewards/src/test/mod.rs

```diff
@@ -246,6 +252,197 @@ fn test_extend_stake_amount() {
+		assert_ok!(StakingRewards::create_reward_pool(Origin::root(), pool_init_config));
+		let (staker, pool_id, amount, duration_preset, total_rewards, total_shares, claim) = (ALICE, StakingRewards::pool_count(), 100_500u32.into(), ONE_HOUR, 100, 200, 50);
+
+		let rewards_pool = StakingRewards::pools(StakingRewards::pool_count()).expect("rewards_pool expected");
+		let staked_asset_id = rewards_pool.asset_id;
+		<<Test as crate::Config>::Assets as Mutate<<Test as frame_system::Config>::AccountId>>::mint_into(staked_asset_id, &staker, amount * 2).expect("an asset minting expected");
+		<<Test as crate::Config>::Assets as Mutate<<Test as frame_system::Config>::AccountId>>::mint_into(staked_asset_id, &StakingRewards::pool_account_id(&pool_id), amount * 2).expect("an asset minting expected");
+		for (asset_id, _) in rewards_pool.rewards.iter() {
+			<<Test as crate::Config>::Assets as Mutate<<Test as frame_system::Config>::AccountId>>::mint_into(*asset_id, &StakingRewards::pool_account_id(&pool_id), amount * 2).expect("an asset minting expected");
+		}
+
+		let mut rewards_pool = StakingRewards::pools(pool_id).expect("rewards_pool expected");
+		let mut inner_rewards = rewards_pool.rewards.into_inner();
+		for (_asset_id, reward) in inner_rewards.iter_mut() {
+			reward.total_rewards += total_rewards;
+		}
+		rewards_pool.rewards = inner_rewards.try_into().expect("rewards expected");
+		rewards_pool.total_shares = total_shares;
+		RewardPools::<Test>::insert(pool_id, rewards_pool);
+
+		assert_ok!(StakingRewards::stake(Origin::signed(staker), pool_id, amount, duration_preset));
+		let stake_id = StakingRewards::stake_count();
+		assert_eq!(<<Test as crate::Config>::Assets as Inspect<<Test as frame_system::Config>::AccountId>>::balance(staked_asset_id, &staker), amount);
+
+		let mut stake = StakingRewards::stakes(stake_id).expect("stake expected");
+		let unlock_penalty = stake.lock.unlock_penalty;
+		let stake_duration = stake.lock.duration;
+		let reductions = stake.reductions.try_mutate(|inner: &mut BTreeMap<_, _>| {
+			for (_asset_id, inflation) in inner.iter_mut() {
+				*inflation -= claim;
+			}
+		}).expect("reductions expected");
+		stake.reductions = reductions;
+		Stakes::<Test>::insert(stake_id, stake);
+
+		let second_in_milliseconds = 1000;
+		Timestamp::set_timestamp(Timestamp::now() + stake_duration * second_in_milliseconds + second_in_milliseconds);
+		assert_ok!(StakingRewards::unstake(Origin::signed(staker), stake_id));
+		assert_eq!(StakingRewards::stakes(stake_id), None);
+		assert_last_event::<Test, _>(|e| {
+			matches!(
+				e.event,
+				Event::StakingRewards(crate::Event::Unstaked { owner, position_id })
+					if owner == staker && position_id == stake_id
+			)
+		});
+
+		let rewards_pool = StakingRewards::pools(StakingRewards::pool_count()).expect("rewards_pool expected");
+		assert_eq!(<<Test as crate::Config>::Assets as Inspect<<Test as frame_system::Config>::AccountId>>::balance(staked_asset_id, &staker), amount * 2);
```

Comment:
Why does the amount double here?

Reply:
Because we minted `amount * 2` here https://github.com/ComposableFi/composable/pull/1330/files#diff-4d413e565fe25c7ab035bacaa5c317ff09dcbc6652e685a2d52b57717ce44a2aR394